spec/requests/dfe_analytics_spec.rb

```diff
@@ -3,13 +3,49 @@
 require "rails_helper"
 
 RSpec.describe "DfE Analytics", type: :request do
-  before { Flipper.enable(Feature::DFE_ANALYTICS_ENABLED) }
+  let(:current_lead_provider) { create(:lead_provider) }
 
-  it "does send DFE Analytics web request events" do
-    expect { get root_path }.to have_sent_analytics_event_types(:web_request)
+  context "when DfE Analytics is enabled" do
+    before { Flipper.enable(Feature::DFE_ANALYTICS_ENABLED) }
```

Comment:
Is it possible to either

a) Just stub this, ie either 

```ruby
before { allow(Feature).to receive(:dfe_analytics_enabled?).and_return(true) }
```

or possibly

```ruby
before do
  allow(Flipper).to receive(:enabled).and_call_original
  allow(Flipper).to receive(:enabled).with(Feature::DFE_ANALYTICS_ENABLED).and_return(true)
end
```

b) or failing that, put it back to how it was after in an `after` block - otherwise this is state which will persist across tests leading to flaky behaviour

Reply:
I don't think this is necessary - since v1.2.0, Flipper uses a memory adapter in the test environment (https://www.flippercloud.io/docs/testing), so each test is run with a fresh instance of Flipper and all features disabled.